src/types/ResponseTypes.ts

```diff
@@ -20,35 +20,77 @@ export enum StatusState {
 export type YaptideResponse = {
 	message: string;
 };
-export const _orderedInputFilesNames = [
+export const _orderedShInputFilesNames = [
 	'info.json',
 	'geo.dat',
-	'mat.dat',
 	'beam.dat',
 	'detect.dat',
+	'mat.dat',
 	'sobp.dat'
 ] as const;
 
-export const _defaultInputFiles = {
+export const _defaultShInputFiles = {
 	'geo.dat': '',
 	'beam.dat': '',
 	'detect.dat': '',
 	'mat.dat': ''
 } as const;
 
-export function isKnownInputFile(name: string): name is SimulationInputFilesNames {
-	return _orderedInputFilesNames.includes(name as SimulationInputFilesNames);
+export const _orderedTopasInputFilesNames = [
+	'info.json',
+	'topas_config.txt'
+] as const;
+
+export const _defaultTopasInputFiles = {
+	'topas_config.txt': '',
+} as const;
+
+export const _orderedFlukaInputFilesNames = [
+	'info.json',
+	'fl_sim.inp'
+] as const;
+
+export const _defaultFlukaInputFiles = {
+	'fl_sim.inp': '',
+} as const;
+
+export function isKnownInputFile(name: string): name is ShInputFilesNames {
+	return _orderedShInputFilesNames.includes(name as ShInputFilesNames);
 }
-export type SimulationInputFilesNames = (typeof _orderedInputFilesNames)[number];
-export type RequiredInputFilesNames = Exclude<SimulationInputFilesNames, 'info.json' | 'sobp.dat'>;
-
-export type SimulationInputFiles = {
-	[Key in RequiredInputFilesNames]: string;
-} & {
-	[Key in Exclude<SimulationInputFilesNames, RequiredInputFilesNames>]?: string;
-} & {
-	[Key in Exclude<string, SimulationInputFilesNames>]: string;
-};
+export type ShInputFilesNames = (typeof _orderedShInputFilesNames)[number];
+export type TopasInputFilesNames = (typeof _orderedTopasInputFilesNames)[number];
+export type FlukaInputFilesNames = (typeof _orderedFlukaInputFilesNames)[number];
+
+export type RequiredInputFilesNames = Exclude<ShInputFilesNames, 'info.json' | 'sobp.dat'>;
+export type RequiredTopasInputFilesNames = Exclude<TopasInputFilesNames, 'info.json'>;
+export type RequiredFlukaInputFilesNames = Exclude<FlukaInputFilesNames, 'info.json'>;
+
+export type SimulationInputFiles = 
+	|{
+		[Key in RequiredInputFilesNames]: string;
+	} & {
+		[Key in Exclude<ShInputFilesNames, RequiredInputFilesNames>]?: string;
+	} & {
+		[Key in Exclude<string, ShInputFilesNames>]: string;
+	}
+	|{
+		[Key in RequiredTopasInputFilesNames]: string;
+	} & {
+		[Key in Exclude<TopasInputFilesNames, RequiredTopasInputFilesNames>]?: string;
+	} & {
+		[Key in Exclude<string, TopasInputFilesNames>]: string;
+	}
+	|{
+		[Key in RequiredFlukaInputFilesNames]: string;
+	} & {
+		[Key in Exclude<FlukaInputFilesNames, RequiredFlukaInputFilesNames>]?: string;
+	} & {
+		[Key in Exclude<string, FlukaInputFilesNames>]: string;
+	}
+	|{
+		[key: string]: string;
+	};
+;
```

Comment:
```suggestion
export type InputFilesRecord<FileNames extends string, OptionalNames extends string> = Omit<
	| {
			[Key in Exclude<FileNames, OptionalNames>]: string;
	  } & {
			[Key in OptionalNames]?: string;
	  },
	never
>;

type ShInputFilesRecord = InputFilesRecord<ShInputFilesNames, 'info.json' | 'sobp.dat'>;
type TopasInputFilesRecord = InputFilesRecord<TopasInputFilesNames, 'info.json'>;
type FlukaInputFilesRecord = InputFilesRecord<FlukaInputFilesNames, 'info.json'>;

export type SimulationInputFiles =
	| ShInputFilesRecord
	| TopasInputFilesRecord
	| FlukaInputFilesRecord;
```

this also requires a small change in Input files editor file